staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go

```diff
@@ -369,3 +371,13 @@ func Convert_Slice_string_To_v1_IncludeObjectPolicy(in *[]string, out *IncludeOb
 	}
 	return nil
 }
+
+// Convert_Slice_string_To_v1_Preconditions allows converting a URL query parameter value.
+func Convert_Slice_string_To_v1_Preconditions(in *[]string, out *Preconditions, s conversion.Scope) error {
+	if len(*in) > 0 {
+		out = NewUIDPreconditions((*in)[0])
```

Comment:
@liggitt for input - is that what we want?
This is explicitly ignoring ResourceVersion that one can potentially want to set.

How this should be handled?

Reply:
it's definitely not correct to take the whole `preconditions` parameter as a uid string. if we're going to decode this into preconditions, we have to have some sort of encoding that lets us address all the preconditions fields (uid, resourceVersion, etc)

Reply:
what does the following output today?
```
parameterCodec.EncodeParameters(&DeleteOptions{
  Preconditions:&Preconditions{
    UID:"123",
    ResourceVersion:"234",
  },
})
```

Reply:
whatever we end up with here has to be symmetric with EncodeParameters, and if EncodeParameters already outputs things for preconditions when converting DeleteOptions to url.Values

answering my own question:
```go
	uid := types.UID("myuid")
	resourceVersion := "myrv"
	var scheme = runtime.NewScheme()
	metav1.AddToGroupVersion(scheme, schema.GroupVersion{Group: "meta.k8s.io", Version: "v1"})
	var parameterCodec = runtime.NewParameterCodec(scheme)
	values, err := parameterCodec.EncodeParameters(&metav1.DeleteOptions{
		Preconditions: &metav1.Preconditions{
			UID:             &uid,
			ResourceVersion: &resourceVersion,
		},
	}, schema.GroupVersion{Group: "meta.k8s.io", Version: "v1"})
	fmt.Printf("%#v, %#v", values, err)
```

> ```
> url.Values{"resourceVersion":[]string{"myrv"}, "uid":[]string{"myuid"}}, <nil>
> ```

so the current implementation of EncodeParameters effectively flattens the nested parameters

Reply:
I think we want something more like this:

```go
func Convert_url_Values_To_v1_DeleteOptions(in *url.Values, out *DeleteOptions, s conversion.Scope) error {
	if err := autoConvert_url_Values_To_v1_DeleteOptions(in, out, s); err != nil {
		return err
	}

	uid := types.UID("")
	if values, ok := in["uid"]; ok && len(values) > 0 {
		uid = types.UID(values[0])
	}

	resourceVersion := ""
	if values, ok := in["resourceVersion"]; ok && len(values) > 0 {
		resourceVersion = values[0]
	}

	if len(uid) > 0 || len(resourceVersion) > 0 {
		if out.Preconditions == nil {
			out.Preconditions = &Preconditions{}
		}
		if len(uid) > 0 {
			out.UID = &uid
		}
		if len(resourceVersion) > 0 {
			out.ResourceVersion = &resourceVersion
		}
	}
	return nil
}
```

Reply:
Roughly.
Plus I think we want to add "+k8s:conversion-gen=false" for the field to avoid autogenerating any conversion for that field.

Reply:
"+k8s:conversion-gen=false" on DeleteOptions.Preconditions?
I'm not sure whether Convert_Xxx_To_v1_Preconditions is needed now.

Reply:
> I'm not sure whether Convert_Xxx_To_v1_Preconditions is needed now.

It is not

Reply:
Exactly - that's why I'm saying to add this conversion-gen=false for it.

Reply:
no-auto generated change in zz_generated.conversion.go will cause verify test failed, hack/update-generated-protobuf.sh also reset these changes.
I have no idea on how to fix it.

Reply:
let it generate autoConvert_url_Values_To_v1_DeleteOptions, and put the Convert_url_Values_To_v1_DeleteOptions in the non-generated conversion.go file

Reply:
get, thanks